docs/contributor/migration-guide.md

```diff
@@ -8,78 +8,164 @@ It covers all necessary changes and deprecations to ensure a smooth transition.
 
-1. Module Team releases a new module version in the GitHub release.
-2. Module Team configures the version to be released in the module config (in the new format) in the internal module-manifest repo.
-3. Module Team updates the version in the related channel in ModuleReleaseMeta.
-4. The submission pipeline gets triggered.
-5. After certain quality gates are passed, the new version of ModuleTemplate gets provisioned into KCP first.
-6. The updated ModuleReleaseMeta gets provisioned into KCP.
-7. Outdated ModuleTemplate (the version not mentioned in ModuleReleaseMeta) should be removed. This step allows both outdated and new versions to coexist temporarily. KLM only handles the version defined in ModuleReleaseMeta.
+```bash
+modulectl -h                # general help
+modulectl create -h         # help for 'create'
+modulectl scaffold -h       # help for 'scaffold'
+```
+
+### 1.2 Command & Flag Differences
+
+#### 1.2.1 Command Mapping
+
+| Operation                | Kyma CLI                       | `modulectl`                         |
+|--------------------------|--------------------------------|-------------------------------------|
+| Scaffold module template | N/A                            | `modulectl scaffold ...`            |
+| Create ModuleTemplate CR | `kyma alpha create module ...` | `modulectl create -c <config-file>` |
+| List installed modules   | `kyma alpha module list ...`   | N/A                                 |
+| Command-specific help    | `kyma alpha module <cmd> -h`   | `modulectl <cmd> -h`                |
+
+#### 1.2.2 Flag & Behavior Differences
+
+| Feature                       | Deprecated in Kyma CLI                            | New in `modulectl`                                             |
+| ----------------------------- | ------------------------------------------------- | -------------------------------------------------------------- |
+| Config-file flag              | `--module-config-file`                            | `--config-file`, shortcut `-c` (all commands)                  |
+| Archive overwrite             | `--module-archive-version-overwrite`              | **Removed** (module versions are immutable)                    |
+| ModuleTemplate naming pattern | `.metadata.name = <module-name>-<channel>`        | `.metadata.name = <module-name>-<version>`                     |
+| Version label                 | `operator.kyma-project.io/module-version` label   | Populated in `.spec.version`                                   |
+| Documentation annotation      | `operator.kyma-project.io/doc-url` annotation     | Defined in `.spec.info.documentation` via `documentation:` key |
+| Manifest & DefaultCR source   | Local file references only                        | Fetched directly from GitHub release URLs                      |
+| Release channel in config     | Required `.channel` field in `module-config.yaml` | **Removed**; channel mapping managed by ModuleReleaseMeta CR   |
```

Comment:
I think I would keep this secton about flags only, and move these differences in the module-config and the generated ModuleTemplate to `2.2` and the new section proposed below.